fastapi/applications.py

```diff
@@ -43,59 +43,608 @@
+
+                Read more about it in the
+                [FastAPI docs for Query Parameters and String Validations](https://fastapi.tiangolo.com/tutorial/query-params-str-validations/#exclude-from-openapi).
+                """
+            ),
+        ] = True,
+        swagger_ui_parameters: Annotated[
+            Optional[Dict[str, Any]],
+            Doc(
+                """
+                Parameters to configure Swagger UI, the autogenerated interactive API
+                documentation (by default at `/docs`).
+
+                Read more about it in the
+                [FastAPI docs about how to Configure Swagger UI](https://fastapi.tiangolo.com/how-to/configure-swagger-ui/).
+                """
+            ),
+        ] = None,
+        generate_unique_id_function: Annotated[
+            Callable[[routing.APIRoute], str],
+            Doc(
+                """
+                Customize the function used to generate unique IDs for the *path
+                operations* shown in the generated OpenAPI.
+
+                This is particularly useful when automatically generating clients or
+                SDKs for your API.
+
+                Read more about it in the
+                [FastAPI docs about how to Generate Clients](https://fastapi.tiangolo.com/advanced/generate-clients/#custom-generate-unique-id-function).
+                """
+            ),
+        ] = Default(generate_unique_id),
+        separate_input_output_schemas: Annotated[
+            bool,
+            Doc(
+                """
+                Whether to generate separate OpenAPI schemas for request body and
+                response body when the results would be more precise.
+
+                This is particularly useful when automatically generating clients.
+
+                For example, if you have a model like:
+
+                ```python
+                from pydantic import BaseModel
+
+                class Item(BaseModel):
+                    name: str
+                    tags: list[str] = []
+                ```
+
+                When `Item` is used for input, a request body, `tags` is not required,
+                the client doesn't have to provide it.
+
+                But when using `Item` for output, for a response body, `tags` is always
+                available because it has a default value, even if it's just an empty
+                list. So, the client should be able to always expect it.
+
+                In this case, there would be two different schemas, one for input and
+                another one for output.
+                """
+            ),
+        ] = True,
+        **extra: Annotated[
+            Any,
+            Doc(
+                """
+                Extra keyword arguments to be stored by FastAPI, not used anywhere.
+                """
+            ),
+        ],
     ) -> None:
-        self.debug = debug
+        """
+        ---
+        """
```

Comment:
@pawamoy I'm noticing that to render the `__init__()` parameters I need to have some non-empty docstring in the `__init__()` method, even if there's already a class docstring...

Am I doing something wrong in the configs or is it expected?

Reply:
I noticed that in the PR yes. That's probably an oversight, I'll take a look :)

Reply:
This is now fixed by latest versions of Griffe and griffe-typingdoc :smile:

Reply:
Thank you! ...although I can't seem to make it work yet 🤔

Reply:
Can you tell me what is not working? I tried locally to remove the docstring from `fastapi.FastAPI.__init__` and the parameters are correctly inserted into the class docs.